src/cmd-artifact-disk

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3 -u
+import argparse
+import logging as log
+import os.path
+import shutil
+import pydoc
+import sys
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, f"{cosa_dir}/cosalib")
+sys.path.insert(0, cosa_dir)
+
+from cosalib.cmdlib import get_basearch
+
+try:
+    from cosalib.build import _Build
+except ImportError:
+    # We're in the container and can't sense the cosa path
+    sys.path.insert(0, '/usr/lib/coreos-assembler/cosalib')
+    from cosalib.build import _Build
+
+IMG_CLASS_MAP = {
+    "aliyun": "cosalib.qemu.AliyunImage",
+    "aws": "cosalib.qemu.AWSImage",
+    "azure": "cosalib.qemu.AzureImage",
+    "dasd": None,
+    "gcp": None,
+    "installer": None,
+    "live": None,
+    "metal": None,
+    "openstack": "cosalib.qemu.OpenStackImage",
+    "qemu": None,
+    "vmware": None,
+}
+
+def set_logger(level):
+    """
+    Set the log level
+    :param level: set the log level
+    :type str
+    """
+    sl = log.DEBUG
+    if level == "error":
+        sl = log.ERROR
+    elif level == "warn":
+        sl = log.WARN
+    elif level == "info":
+        sl = log.INFO
+
+    log.basicConfig(format='[%(asctime)s  %(levelname)s]: %(message)s',
+                    level=sl)
+
+
+def get_build_class(imgtype, build_root, build="latest"):
+    mod_name = IMG_CLASS_MAP.get(imgtype)
+    if mod_name is None:
+        raise NotImplementedError(f"target {imgtype} has not been ported yet")
+
+    log.info(f"Loading Image Class for {mod_name}")
+    mod = pydoc.locate(mod_name)
```

Comment:
Why not `importlib`?

```python3
>>> import importlib            
>>> mod = importlib.import_module('os.path')                         
>>> print(mod)                  
<module 'posixpath' from '/usr/lib64/python3.7/posixpath.py'>
>>> import os.path              
>>> print(os.path)              
<module 'posixpath' from '/usr/lib64/python3.7/posixpath.py'>
```

Reply:
Unresolved as this is still a question I have :-)

Reply:
Indeed. I tested it:
```
[root@e09b8408ba4a src]# ./cmd-artifact-disk target metal
Traceback (most recent call last):
  File "./cmd-artifact-disk", line 124, in <module>
    artifact_cli().build_artifacts()
  File "./cmd-artifact-disk", line 103, in artifact_cli
    force=args.force)
  File "./cmd-artifact-disk", line 33, in get_build_class
    mod = importlib.import_module(mod_name)
  File "/usr/lib64/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 962, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cosalib.metal.MetalVariantImage'; 'cosalib.metal' is not a package
```

Reply:
Given there's only one class now, is there really a need for the dynamic loading versus just doing it unconditionally?

Reply:
I think not...and dropped. I had plumbed it for the metal stuff I've been working on, but I think that the justification for the complexity is tenuous at best. 

I pushed a new commit that uses direct imports.